### PR TITLE
Fixes #28627 - update entitlements report due to added macro

### PR DESCRIPTION
**Pull request by domitea**
I've added a `host_collections` macro in https://github.com/Katello/katello/pull/8586 So I have to update report template also.

### CODE REVIEW

**Comment by xprazak2** [comment]
Works for me, maybe the new macro could be used in [Inventory template](https://github.com/theforeman/community-templates/blob/develop/report_templates/ansible_inventory.erb#L161) as well.

**Review by ares — COMMENTED** [review]
I'm not against, however could this have been solved by `host.host_collection.map { |c| c.to_s }`? The new macro shields us from AR interface, which is great, however it's misleading a bit, because it's not clear it returns string representation of objects. Perhaps if it's call `host_collections_names` it would be better.

**Comment by xprazak2** [comment]
What is the status here? @domitea, do you think this could be solved just by using `map` and therefore no changes in katello would be needed?

**Comment by domitea** [comment]
@ares, @xprazak2 Yes, same behavior can be achieved by `.map(...) { ... }` call but I am fan of hiding core libraries like AR from user as far as I can to avoid possibility of usage some future security issue (like SQL injection attack for non escaped SQL query). So I've updated templates according to new macros created.

**Comment by xprazak2** [comment]
Works as expected with the new macros in katello.

File: report_templates/ansible_inventory.erb
Comment (jturel) [inline report_templates/ansible_inventory.erb]: I'm not sure if this is an equivalent change, but I'm unsure how to test this template. The value is now a comma separated string of names rather than an array of names. If this change isn't required I'd rather omit it.

Comment (domitea) [reply report_templates/ansible_inventory.erb]: I think that my addition is correct because actually we're trying to show multiple values in one cell of table. Also, this template is report template so it's intended for human, not machine (like job template) and that means that multiple value should have a delimiter.

Comment (ezr-ondrej) [reply report_templates/ansible_inventory.erb]: That's not entirely true @domitea. If I select json format as an output format, I might very likely intend to consume it by machine for further processing. That being said, we might want to include formaters for columns that would take care of different formatting of columns. what actually goes over this PR. But I believe the macro should return an array of values and templating take care of the formatting :thinking:

Comment (tbrisker) [reply report_templates/ansible_inventory.erb]: I agree too, macros shouldn't handle formatting. also comma-seperated list might mess up csv reports (not sure if commas in field values are escaped when rendering?)

**Comment by domitea** [comment]
@jturel I've removed change in ansible inventory template. Is this good for you?

**Review by jturel — APPROVED** [review]
ACK

Thank you @domitea :+1:

**Comment by tbrisker** [comment]
any further concerns @xprazak2 @ares or @ezr-ondrej ? I'd like to get this one in today if all your concerns have been addressed.

**Comment by xprazak2** [comment]
I have no additional concerns.

**Comment by ares** [comment]
Do we need to bump katello version in metadata/requirements? Iiuc we use new macros. Also after this is merged, we need to sync templates to core. Other than that, all green

**Comment by tbrisker** [comment]
> Do we need to bump katello version in metadata/requirements? Iiuc we use new macros. Also after this is merged, we need to sync templates to core. Other than that, all green

Good point, I think this is worth cherry-picking to older katello versions but we'll need to get ack from katello folks about which version will be the minimal this will land in.

**Comment by ares** [comment]
FTR I'd be ok with merging this as is and adding metadata tomorrow or after katello grooming.

**Comment by ezr-ondrej** [comment]
> any further concerns @xprazak2 @ares or @ezr-ondrej ? I'd like to get this one in today if all your concerns have been addressed.

My (originally @jturel's) formatting comment https://github.com/theforeman/community-templates/pull/693#discussion_r390240101 got ignored out. have we tested it in json? Are we ok with it not being array?
The escaping should happen though, so it's just UX (usability) issue. I'm ok with solving it as follow up later.
Thus I'm ok if it get merged.

**Review by tbrisker — APPROVED** [review]
ok, based on all comments i'll go ahead and merge this for now, thanks @domitea and please open followup PRs (using "Refs #28627..." commit message) to address the concerns raised.

**Comment by ares** [comment]
@domitea to be explicit, two concerns I understood:
1) we need to set minimum version of katello if we rely on new macros
2) I don't know how to fix correct structure other than just passing array of strings to report_row macro, either we need to change macros to return array of strings or this template to simply get objects and do `.map { |s| s.name }`, so macros wouldn't be required and we wouldn't have to update metadata

**Comment by ezr-ondrej** [comment]
@domitea any progress with followup PR?

**Pull request merged**
